mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -463,10 +462,155 @@ void vector::MultiDimReductionOp::build(OpBuilder &builder,
   build(builder, result, kind, source, acc, reductionDims);
 }
 
+/// Computes the result of reducing a constant vector where the accumulator
+/// value, `acc`, is also constant.
+template <typename T>
+static OpFoldResult computeConstantReduction(T src, T acc, int64_t times,
+                                             CombiningKind kind,
+                                             ShapedType dstType);
+template <typename T>
+static T power(const T &a, int64_t times);
```

Comment:
[nit] `power` -> `computePowerOf` (or similar).

From https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly:

>Function names should be verb phrases (as they represent actions), and command-like function should be imperative. The name should be camel case, and start with a lower case letter (e.g. openFile() or isFoo()).

Also, these hooks don't really belong here. "APInt.h" and "APFloat.h" seem like more suitable place. Could you add a TODO to move them? Or, better still, do it within this PR.

Reply:
I do have commit access.
Changed the name. Arithmetic functions are quite established. C++ has  std::{min/max/pow/log/..} , LLVM already has llvm::{minimum,maxnum} rather than `computeMaximum`, computeAdd etc. : https://llvm.org/doxygen/classllvm_1_1APFloat.html It seems weird to single out power as "computePowerOf".